Fix transport logging

ersonp opened the pull request:
Did you run `make format && make check`?

Fixes #1363	
Fixes #1388

 Changes:	
- Fixed failing tests
- Fixed tp log failure on second start
- Fixed log data reset (now the data is persisted between visor restarts)

How to test this PR:
1. To test interday logs change [`2006-01-02`](https://github.com/ersonp/skywire/blob/1ed23d0fa6b12efb1c633138bb909492b5e278d6/pkg/transport/log.go#L23) to this `2006-01-02--3:4`
2. Build `make build`
3. Check the data in the new files that are created resets or not
4. Revert the changes and build again `make build`
5. Create CSV files older than 7 days in `./local/transport_logs` eg `2022-10-08.csv` (yyyy-mm-dd)
6. Check if they are deleted on visor startup
7. Start a vpn-server
8. Start a vpn-client and connect to it
9. Check logs (check if the logs sent and received bytes in csv reset every day (minute) file)
10. Stop visor
11. Restart the vpn-client and connect to the server again
12. Check logs (test with at least 2 transports and 2 different servers)
13. Run `./skywire-cli config gen -ir` check the location of `transport_logs` and `custom`
14. Run `./skywire-cli config gen -irp` check the location of `transport_logs` and `custom`
15. Run `./skywire-cli config gen -iru` check the location of `transport_logs` and `custom`

ersonp committed:
Fix writeToCSV logic

ersonp committed:
Fix readFromCSV

This commit fixes readFromCSV by updating the method today to todayFileName which also gives the .csv file extension along with the file name.

ersonp committed:
Minor fix

ersonp committed:
Add mutex lock

ersonp committed:
Change Timestamp type to unix

ersonp committed:
Fix FileTransportLogStore logic

This commit fixes two things; csv update row issue when update is smaller in size than the orignal and the second issue where we used to overwrite and reset LogEntry after a restart.

Now whenever a new log entry is created with MakeLogEntry we read the Entry from the log store and if there is an entry then we add it to the new one which means the update will always be equal to or bigger than the old entry in the csv which will prevent the issue.

The second part is that now we nolonger will reset the transport data on restart and the data will now be persisted and loaded back if it's available.

ersonp commented:
@0pcom this works fine assuming everything is being done within one day. What should be done for multiday logs? 
- reset the counter every day
OR
- continue from the previous day

mrpalide reviewed (approved):
Work well.

0pcom commented:
> @0pcom this works fine assuming everything is being done within one day. What should be done for multiday logs?
> 
> * reset the counter every day
>   OR
> * continue from the previous day

the csv filename is dated for the current day, right?

I think the only way to do it is to assume that we dont fetch the logs for the current day on that day, to avoid fetching logs for only part of the day depending on the time zone.

Basically, if the logs can always be written to a file for the current day.. it only gets difficult if we try to reconcile logs which, depending on the timezone and node location, fall on different days.

 it would actually be best to use a common time. Then we dont have a continuous skew across different logs based on timezone.

but in terms of multiday and how long we keep logs.. that is what log rotation interval is for, which i think you already added to the config.

* delete the logs which are older than specified in logrotation interval

* set default log rotation interval to either 14 or 28 days.. or at least 7 days, depending how big these files may get

* for transports that cross the 'day line' we should log the amount of bandwidth used on the previous day until the start of the next day, then start logging the same transport on the next day as soon as more bandwidth is sent or recieved in a new log file.

ersonp committed:
Fix service discover error

After the recent changes we were sending empty info{} to the SD that is not updated which caused a json error.

Although we have set it as omitempty it stil sent empty json because VPNInfo was not a pointer.

We fix it by having VPNInfo as a pointer in Service inside pkg servicedisc and we also set the fields in VPNInfo to be omitempty.

ersonp committed:
Fix inter-day transport logging

This commit fixes the interday transport logging by resetting the LogEntry if it's a new day.

This is achieved by adding a Reset method to LogEntry first. Then we add field fileName to fileTransportLogStore check for a new day.

When a new day is detected inside the method Record insted of saving the entry to a new file of the new day we save it to the file of the old day instead. The timestamp of the entry will be of the new day indicating that the transport log was inter-day.

After we have saved this entry to the old log file we reset the LogEntry so that from the next cycle we start from zero when we are saving to the new file of today. We also set the fileName with todayFileName to indicate that the new logs can now be written to todays file.

ersonp committed:
Fix method Reset of LogEntry

This commit fixes the the method Reset by changeing the type of RecvBytes and SentBytes from uint64 to *uint64.

The issue was caused because we were resetting the copies of the fields so the reset was never regestered outside of the method.

It also contains minor fixes for everything to work with the pointers.

ersonp committed:
Add field LogStore back to Transport in config

The field LogStore was unused after the refactore because we were only using the mem log store.

We bring back the field with Type and Location and also move the filed LogRotationInterval inside LogStore renaming it to RotationInterval.

We set the default for RotationInterval to be 7 days and we use the data from LogStore to select the Logstore Type as well as the LogStore Location.

ersonp committed:
Add CustomDmsgHTTPPath to config

ersonp commented:
Deleting old logs with log rotation intervals is the last thing remaining.

ersonp committed:
Clean logs older than log interval

ersonp committed:
Minor fixes

ersonp committed:
Fix panics in tests

ersonp committed:
Update changelog

ersonp committed:
Minor test fix

ersonp committed:
Minor change

ersonp committed:
Minor change
